client/landing/stepper/declarative-flow/internals/steps-repository/components/accent-color-control/index.tsx

```diff
@@ -32,40 +33,56 @@ enum COLORS {
 	VividPurple = '#9B51E0',
 }
 
-const COLOR_OPTIONS = [
-	{
-		label: 'Lettre',
-		value: COLORS.Lettre,
-		icon: <ColorSwatch color={ COLORS.Lettre } />,
-	},
-	{
-		label: 'Black',
-		value: COLORS.Black,
-		icon: <ColorSwatch color={ COLORS.Black } />,
-	},
-	{
-		label: 'Vivid red',
-		value: COLORS.VividRed,
-		icon: <ColorSwatch color={ COLORS.VividRed } />,
-	},
-	{
-		label: 'Vivid purple',
-		value: COLORS.VividPurple,
-		icon: <ColorSwatch color={ COLORS.VividPurple } />,
-	},
-	{
-		label: 'Custom',
-		value: 'custom',
-		icon: <Icon className="custom_color_icon" icon={ color } width={ 22 } height={ 22 } />,
-	},
-];
+const getColorOptions = ( shouldLimitGlobalStyles: boolean ) => {
+	const locked = shouldLimitGlobalStyles && {
+		extra: <Gridicon icon="lock" size={ 18 } className="extra-gridicon" />,
+	};
+
+	return [
+		{
+			label: 'Lettre',
+			value: COLORS.Lettre,
+			icon: <ColorSwatch color={ COLORS.Lettre } />,
+		},
+		shouldLimitGlobalStyles && {
+			label: 'Unlock more colors with a Premium plan',
+			value: 'Unlock more colors with a Premium plan',
+			isLabel: true,
+		},
+		{
+			label: 'Black',
+			value: COLORS.Black,
+			icon: <ColorSwatch color={ COLORS.Black } />,
+			...locked,
+		},
+		{
+			label: 'Vivid red',
+			value: COLORS.VividRed,
+			icon: <ColorSwatch color={ COLORS.VividRed } />,
+			...locked,
+		},
+		{
+			label: 'Vivid purple',
+			value: COLORS.VividPurple,
+			icon: <ColorSwatch color={ COLORS.VividPurple } />,
+			...locked,
+		},
+		{
+			label: 'Custom',
+			value: 'custom',
+			icon: <Icon className="custom_color_icon" icon={ color } width={ 22 } height={ 22 } />,
+			...locked,
+		},
+	].filter( ( item ) => !! item );
```

Comment:
Nitpick: I find this `filter` call confusing. You can avoid it entirely doing this:

```js
return [
	// ...
	...	shouldLimitGlobalStyles ? [ {
		label: 'Unlock more colors with a Premium plan',
		value: 'Unlock more colors with a Premium plan',
		isLabel: true,
	} ] : [],
	// ...
];
```

Reply:
I knew it could be done somehow with destructuring, thanks!